components/Document/DocumentHeader.tsx

```diff
@@ -53,77 +55,55 @@ function DocumentHeader({
     userVote,
     score,
     hubs,
+    id: documentID,
   } = document;
 
   const currentAuthor = parseAuthorProfile(currentUser.author_profile);
   const [isAuthorClaimModalOpen, setIsAuthorClaimModalOpen] = useState(false);
-  const [voteState, setVoteState] = useState({
+  const [voteState, setVoteState] = useState<{
+    userVote: GrmVoteType | null | undefined;
+    voteScore: number;
+  }>({
     userVote: userVote,
     voteScore: score,
   });
 
   useEffect(() => {
-    const isSubmittedByCurrentUser =
-      currentUser && currentUser?.id === createdBy?.id;
-    if (isSubmittedByCurrentUser) {
-      setVoteState({
-        ...voteState,
-        userVote: UPVOTE,
-        voteScore: document.score,
-      });
-    } else {
-      setVoteState({
-        ...voteState,
-        userVote: document.userVote,
-        voteScore: document.score,
-      });
-    }
-  }, [currentUser, document]);
-
-  const handleVoteSuccess = ({ voteType, increment }) => {
-    let newVoteScore = voteState.voteScore;
-    if (voteType === UPVOTE) {
-      if (voteState.userVote === DOWNVOTE) {
-        newVoteScore = voteState.voteScore + increment + 1;
-      } else if (voteState.userVote === UPVOTE) {
-        // This shouldn't happen but if it does, we do nothing
-        console.log("User already casted upvote");
-        return;
-      } else {
-        newVoteScore = voteState.voteScore + increment;
-      }
-    } else if (voteType === DOWNVOTE) {
-      if (voteState.userVote === DOWNVOTE) {
-        console.log("User already casted downvote");
-        return;
-      } else if (voteState.userVote === UPVOTE) {
-        newVoteScore = voteState.voteScore + increment - 1;
-      } else {
-        newVoteScore = voteState.voteScore + increment;
-      }
-    }
+    setVoteState({
+      ...voteState,
+      userVote: document?.userVote,
+      voteScore: document.score,
+    });
+  }, [document]);
 
+  const handleVoteSuccess = ({ increment, voteType }) => {
     setVoteState({
       userVote: voteType,
-      voteScore: newVoteScore,
+      voteScore: voteState.voteScore + increment,
     });
   };
 
   let onUpvote, onDownvote;
   if (document.isReady) {
     onUpvote = createVoteHandler({
-      voteType: UPVOTE,
-      unifiedDocument,
       currentAuthor,
-      onSuccess: handleVoteSuccess,
+      currentVote: voteState?.userVote,
+      documentCreatedBy: nullthrows(createdBy),
```

Comment:
We should make sure no error is thrown here if `createdBy` does not exist.
I believe this will throw an error.

We do not want this because throwing an error will likely break the header from rendering in one way or another. There are times where the user may be deleted but their paper is still available and we want to render it.

Reply:
ah got it.
Yeah unfortunately we do delete users from time to time. 
There are legal laws around it.
Glad it won't break the page though.